nilearn/input_data/nifti_masker.py

```diff
@@ -12,7 +13,9 @@
 from .. import _utils
 from .. import image
 from .. import masking
+from nilearn import plotting
```

Comment:
I think we should avoid importing plotting from modules that will generate reports because we will end up creating circular dependencies. rather than have all classes inherit from ReportMixin and use inherited private methods to generate their reports, maybe we can have a public function in reports that calls private methods of the objects to get the content and template name it needs. this way reported objects can completely ignore the reporting module and just need to store the information needed for reporting.
if some common utilities are really needed by these objects, they could go in utils.py or some separate reporting.py module that doesn't depend on plotting or the modules that will use it

Reply:
another reason to keep analysis / computing code and plotting code separate is that the rest of nilearn doesn't require matplotlib

Reply:
and since you will need custom plotting for each reported class, you could have separate functions in a reporting.py module and for example look them up by the class name in a dictionary. it's not very elegant because that's what inheritance does for you, but I would still prefer that to importing plotting everywhere. maybe we can come up with a better solution. also, at some point we may want different kinds of reports for the same class, eg depending on whether it is on its own or part of a pipeline.

Reply:
I wasn't able to keep them totally separate, since the private method of `NiftiMasker` that creates the figure in the report requires plotting, still. But I tried to address this a bit ! Let me know what you think.